Corona-Warn-App/src/main/java/de/rki/coronawarnapp/contactdiary/ui/day/ContactDiaryDayFragment.kt

```diff
@@ -0,0 +1,57 @@
+package de.rki.coronawarnapp.contactdiary.ui.day
+
+import android.os.Bundle
+import android.view.View
+import androidx.fragment.app.Fragment
+import androidx.navigation.fragment.navArgs
+import com.google.android.material.tabs.TabLayoutMediator
+import de.rki.coronawarnapp.R
+import de.rki.coronawarnapp.contactdiary.ui.day.tabs.ContactDiaryDayFragmentsAdapter
+import de.rki.coronawarnapp.contactdiary.util.registerOnPageChangeCallback
+import de.rki.coronawarnapp.databinding.ContactDiaryDayFragmentBinding
+import de.rki.coronawarnapp.util.di.AutoInject
+import de.rki.coronawarnapp.util.ui.observe2
+import de.rki.coronawarnapp.util.ui.viewBindingLazy
+import de.rki.coronawarnapp.util.viewmodel.CWAViewModelFactoryProvider
+import de.rki.coronawarnapp.util.viewmodel.cwaViewModelsAssisted
+import javax.inject.Inject
+
+class ContactDiaryDayFragment : Fragment(R.layout.contact_diary_day_fragment), AutoInject {
+    private val binding: ContactDiaryDayFragmentBinding by viewBindingLazy()
+
+    private val navArgs by navArgs<ContactDiaryDayFragmentArgs>()
+
+    @Inject lateinit var viewModelFactory: CWAViewModelFactoryProvider.Factory
+    private val viewModel: ContactDiaryDayViewModel by cwaViewModelsAssisted(
+        factoryProducer = { viewModelFactory },
+        constructorCall = { factory, _ ->
+            factory as ContactDiaryDayViewModel.Factory
+            factory.create(navArgs.selectedDay)
+        }
+    )
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        binding.contactDiaryDayViewPager.adapter =
+            ContactDiaryDayFragmentsAdapter(this, viewModel.contactDiaryTabs, navArgs.selectedDay)
+
+        TabLayoutMediator(binding.contactDiaryDayTabLayout, binding.contactDiaryDayViewPager) { tab, position ->
+            val tabSource = viewModel.contactDiaryTabs[position]
+            tab.text = resources.getString(tabSource.tabNameResource)
+        }.attach()
+
+        binding.contactDiaryDayViewPager.registerOnPageChangeCallback {
+            viewModel.updateCurrentTab(it)
+        }
+
+        binding.contactDiaryDayFab.setOnClickListener {
+            viewModel.onCreateButtonClicked()
+        }
+
+        viewModel.uiState.observe2(this) {
```

Comment:
tab handling seems kind of wonky to me, any suggestions on how to do this better?

Reply:
i assumed that fabs are supposed to be static on the view of the user and not move when paging, not actually sure if this is 100% true though

Reply:
We could add a little slide out/in animation :smirk: